bsip-00XX Dynamic market fees.md

```diff
@@ -0,0 +1,183 @@
+
+# General notes
+Each asset will have either usual market fee or a table with dynamic market fees that will represent the table above. 
+
+A switch **market_fee_type (REGULAR, DYNAMIC)** should clearly indicate, which market fee type is used.
+
+**Makers** are the ones who put up their orders first.
+
+And **takers** are the ones to put up their orders later in time than makers. 
+
+The following table 
+# Discussion
+
+We have had a discussion what's the best way to average out Trading Volume.
+
+There are a few ways of doing this: 
+
+- Use an **average volume since day the beginning of trading**.  
+Up for discussion is - what to take as a day 1. 1st trade of this asset? 1st trade minus 30 days?  Account open date? 
+This way we have a normal fair average volume if we figure out the fair beginning interval date.
+However, if a trader started trading this asset a few years ago he might be in a significant disadvantage.
+
+- **Average trading volume over the last 30 days - sliding window**.
+This way it is a bit clearer how to calculate. 
+However there might be extreme but real cases where this kind of calculation might not be fair: 
+a. when a person who usually trades high volumes throughout the year goes to vacation for 30 days. Suddenly all his discount is gone when he comes back
+b.   When another person comes in and makes 1 or 2 high volume trades while trading really low volumes before that. Suddenly this person might have a big discount. 
+30-day volume 
+This mehod gives enough information to figure out last 30-days traders habits with much lower memory requirements but slightly less precision than the 30-day average sliding window.
+Average trading volume over the last 365 days (a year) is a variation of this method.
+
+- **30-day volume method**
+This is quite similar to the 30-day average method and shares it's advantages and drawbacks.
+
+# Technical Specifications
+
+We suggest using average total volume for a particular asset. However, it's open for the discussion.
+
+## Statistics Storage
+
+There are two options for storing account statistics for trading:
+
+1. expand class account_statistics_object by adding a map of trade_statistics_object for each asset
+2. create a new type trade_statistics_object and a new index
+
+
+## Statistics Update
+Statistics updates during fill_order. 
+We need to update statistics for the user and for the specific asset at the same time.
+
+ 
+
+## Calculations
+
+### Average since Day 1
+
+**Trader Activity** is estimated as average trading volume of a user for a given asset for 1 day.
+
+`Trader_activity = total_volume / (Now - first_start_date) //(in days).`
+
+This way we accumulate total bought volume for each user for each asset and average out daily volumes for his entire term of trading of this asset (per day).
+
+Volume discounts are estimated based on the average daily `Trader_Activity  * 30 days.`
+
+_Trade statistics object_
+
+
+```
+class trade_statistics_object
```

Comment:
Need many of those per user given that I could trade in USD and CNY ...

Reply:
This will eat up a lot of RAM and, given that it is required for consensus, this also affects block producers

Reply:
pls see comment above.